setup.py

```diff
@@ -1,27 +1,45 @@
-#!/usr/bin/python
+#! /usr/bin/env python
 
-##################
-# setup.py
-#
-# Copyright David Baddeley, 2009
-# d.baddeley@auckland.ac.nz
-#
-# This file may NOT be distributed without express permision from David Baddeley
-#
-##################
+__version__ = '0.1.0'
 
-
-def configuration(parent_package='',top_path=None):
-    from numpy.distutils.misc_util import Configuration
-    config = Configuration('pyoptic2',parent_package,top_path)
-    config.add_subpackage('pyoptic2')
-    
-    #config.make_svn_version_py()  # installs __svn_version__.py
-    #config.make_config_py()
-    #config.get_version('PYME/version.py')
-    return config
+from setuptools import setup, find_packages
+from os import path
 
 if __name__ == '__main__':
-    import setuptools
-    from numpy.distutils.core import setup
-    setup(**configuration(top_path='').todict())
+
+    if sys.version_info[0] == 2:
+        sys.exit("Sorry, Python 2 is not supported")
+
+    this_directory = path.abspath(path.dirname(__file__))
+    with open(path.join(this_directory, 'README.md'), encoding='utf-8') as file_id:
+        long_description = file_id.read()
+    short_description = ('An experimental python-based raytracing/optical CAD package which '
+                         'works from a programmatic definition of the optical system.')
+
+    setup(name='pyoptic2',
+          version=__version__,
+          license="GPLv3",
+          description=short_description,
+          long_description=long_description,
+          long_description_content_type='text/markdown',
+          author="David Baddeley",
+          author_email='david.baddeley.nz@gmail.com',
+          packages=find_packages(),
+          keywords=['Optics', 'Ray Tracing'],
+          url='https://github.com/David-Baddeley/pyoptic2',
+          project_urls={"Bug Tracker": "https://github.com/David-Baddeley/pyoptic2/issues",
+                        "Source Code": "https://github.com/David-Baddeley/pyoptic2g"},
+          classifiers=['Development Status :: 5 - Production/Stable',
+                       'License :: OSI Approved :: GNU General Public License v3 (GPLv3)',
+                       'Programming Language :: Python :: 3.6'],
+          install_requires=['numpy',
+                            'scipy',
+                            'matplotlib',
+                            'mayavi',
```

Comment:
This is only needed for 3D visualisation, and can be exceptionally hard to satisfy in, e.g., a conda environment